openedx/features/subscriptions/management/commands/send_subscriptions_expiry_emails.py

```diff
@@ -0,0 +1,96 @@
+from smtplib import SMTPException
+
+from django.conf import settings
+from django.core.mail import send_mail
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+
+from edxmako.shortcuts import render_to_string
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.features.subscriptions.utils import get_subscription_renew_url
+from openedx.features.subscriptions.models import UserSubscription
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Send subscripion impending expiry and post expiry email notifications.
+    """
+    def _send_email_notifications(self, context_values, subject_filename, message_filename):
+        """
+        Send email notifications from the given context values.
+        """
+        from_address = configuration_helpers.get_value(
+            'email_from_address',
+            settings.DEFAULT_FROM_EMAIL
+        )
+        marketing_site_root = configuration_helpers.get_value('MARKETING_SITE_ROOT')
+        subscriptions_marketing_url = '{marketing_root_url}subscriptions'.format(
+            marketing_root_url='' if not marketing_site_root else marketing_site_root
+        )
+        for subscription_id, user_email in context_values:
+            subscription_renew_url = get_subscription_renew_url(subscription_id)
+            context = {
+                'subscriptions_marketing_url': subscriptions_marketing_url,
+                'subscription_renew_url': subscription_renew_url,
+                'subscription_id': subscription_id,
+                'user_email': user_email,
+            }
+            subscription_expiry_email_subject = render_to_string(subject_filename, {})
+            subscription_expiry_email_subject = ''.join(subscription_expiry_email_subject.splitlines())
+            subscription_expiry_email_message = render_to_string(message_filename, context).lstrip()
+            try:
+                send_mail(
+                    subscription_expiry_email_subject,
+                    subscription_expiry_email_message,
+                    from_address,
+                    [user_email],
+                    fail_silently=False
+                )
+            except SMTPException:
+                logger.warning('Failure sending subscription expiry notification e-mails for user %s', user_email)
+
+    def handle(self, *args, **options):
+        """
+        Send expiry emails to relevant users.
+        """
+        impending_expiry_subscriptions = UserSubscription.objects.filter(
+            Q(expiration_date__isnull=False) & Q(expiration_date=date.today()),
+            Q(expiry_notification_before_expiration_date=False)
+        )
+        impending_expiry_subscriptions_context_values = impending_expiry_subscriptions.values_list(
+            'subscription_id', 'user__email'
+        )
+
+        expired_subscriptions = UserSubscription.objects.filter(
```

Comment:
I think we should remove those `expiry_notification_before_expiration_date` and `expiry_notification_after_expiration_date` fields and just send notification
1. On the day of expiry for impending expiry subscriptions as we doing now
2. One the day after expiry for expired subscriptions one day after they are expired

That would be more scalable approach.

Reply:
Yes I thought that would be an appropriate approach as well but this was mentioned in the acceptance criteria. we could run this command once a day as no subscription is going to change its expiry in a day. This would ensure that the users are not spammed as well.